Source/SerializationTransformer.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SerializationTransformer.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+/// A closure that given a String as input output the transformed String, used to transform keys of the json
+public typealias KeyTransformer = (key: String) -> String
+
+/**
+ *  A protocol that special `Serializable` objects can adopt to transform other `Serializable` objects.
+ *  At the moment only key transforms are supported, see `SnakecaseTransformer` for a concrete implementations
+ */
```

Comment:
`At the moment only key transforms are supported` I think we need a better way to constraint this. For now we only have this line in the documentation and the `promise` that all of our extension in foundation and JSONAPI will use this just for keys, but won't necessarily be always the case.